e2e/e2etests/test_erc20_deposit.go

```diff
@@ -3,14 +3,19 @@ package e2etests
 import (
 	"math/big"
 
+	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/zeta-chain/zetacore/e2e/runner"
 	"github.com/zeta-chain/zetacore/e2e/utils"
+	"github.com/zeta-chain/zetacore/zetaclient/testutils"
 )
 
 func TestERC20Deposit(r *runner.E2ERunner) {
-	hash := r.DepositERC20WithAmountAndMessage(big.NewInt(100000), []byte{})
+	hash := r.DepositERC20WithAmountAndMessage(r.DeployerAddress, big.NewInt(100000), []byte{})
 
 	// wait for the cctx to be mined
 	cctx := utils.WaitCctxMinedByInTxHash(r.Ctx, hash.Hex(), r.CctxClient, r.Logger, r.CctxTimeout)
 	r.Logger.CCTX(*cctx, "deposit")
+
+	// deposit ERC20 to restricted address
+	r.DepositERC20WithAmountAndMessage(ethcommon.HexToAddress(testutils.RestrictedEVMAddressTest), big.NewInt(100000), []byte{})
```

Comment:
The deposit test for restricted address is a bit special. Zetaclient will NOT observe the deposit since the address is blacklisted. I manually looked into the log prints of container `zetaclient0` to make sure the deposits are ignored. You can see the screenshot of log prints in PR https://github.com/zeta-chain/node/pull/1790.

For separate test, see this https://github.com/zeta-chain/node/pull/1791/commits/ce021e113520dbab64e1d27e5512c968b2dcefff